datalake/src/main/java/com/sequenceiq/datalake/service/resize/recovery/ResizeRecoveryService.java

```diff
@@ -0,0 +1,116 @@
+package com.sequenceiq.datalake.service.resize.recovery;
+
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.NotImplementedException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.dto.NameOrCrn;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.response.recovery.RecoveryStatus;
+import com.sequenceiq.datalake.entity.DatalakeStatusEnum;
+import com.sequenceiq.datalake.entity.SdxCluster;
+import com.sequenceiq.datalake.entity.SdxStatusEntity;
+import com.sequenceiq.datalake.flow.SdxReactorFlowManager;
+import com.sequenceiq.datalake.flow.chain.DatalakeResizeFlowEventChainFactory;
+import com.sequenceiq.datalake.service.recovery.RecoveryService;
+import com.sequenceiq.datalake.service.sdx.SdxService;
+import com.sequenceiq.datalake.service.sdx.status.SdxStatusService;
+import com.sequenceiq.flow.core.Flow2Handler;
+import com.sequenceiq.flow.domain.FlowLog;
+import com.sequenceiq.flow.service.flowlog.FlowChainLogService;
+import com.sequenceiq.sdx.api.model.SdxRecoverableResponse;
+import com.sequenceiq.sdx.api.model.SdxRecoveryRequest;
+import com.sequenceiq.sdx.api.model.SdxRecoveryResponse;
+
+@Component
+/**
+ * Provides entrypoint for recovery of failed SDX resize.
+ *
+ * The main entry point is {@code triggerRecovery}, which starts a cloudbreak Flow to recover the Data Lake.
+ * To ensure a Resize recovery is appropriate, use one of {@code canRecover} or {@code validateRecovery}.
+ */
+public class ResizeRecoveryService implements RecoveryService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ResizeRecoveryService.class);
+
+    @Inject
+    private SdxService sdxService;
+
+    @Inject
+    private SdxStatusService sdxStatusService;
+
+    @Inject
+    private SdxReactorFlowManager sdxReactorFlowManager;
+
+    @Inject
+    private Flow2Handler flow2Handler;
+
+    @Inject
+    private FlowChainLogService flowChainLogService;
+
+    /**
+     * Determines if it's possible to run a <em>Resize</em> recovery, returning a simple boolean indicator
+     *
+     * @param crn       the CRN of the user requesting recovery
+     * @param nameOrCrn the name or CRN of the resource to recover
+     * @return true if resize recovery is possible.
+     */
+    public boolean canRecover(String crn, NameOrCrn nameOrCrn) {
+        SdxCluster cluster = sdxService.getByNameOrCrn(crn, nameOrCrn);
+        FlowLog flowLog = flow2Handler.getFirstStateLogfromLatestFlow(cluster.getId());
+        if (DatalakeResizeFlowEventChainFactory.class.getSimpleName().equals(flowChainLogService.getFlowChainType(flowLog.getFlowChainId()))) {
+            SdxStatusEntity actualStatusForSdx = getSdxStatusEntity(crn, nameOrCrn);
```

Comment:
I feel like this can be simplified since you already have the SdxCluster, otherwise you are technically getting the cluster a second time for no real reason through this method call.